### PR TITLE
Remove grant on data node bootstrap

### LINKED ISSUE #3957: [Bug]: Internal handling of ACL needs improvement
### What type of bug is this?

Configuration

### What subsystems and features are affected?

Other

### What happened?

When bootstrapping a data node, a new database is optionally created with default permissions.

In PG15 non-superuser accounts are not allowed to create tables in public schemas unless they are owners. This lead to a bunch of test failures, which was handled in #3941 by granting full access to the public schema on the data node. Since this might not be the intention of the database administrator, it can result in permissions that are to liberal.

In general, we should avoid granting liberal permissions by default and instead rely on default permissions, since these are set by the administrator.

### TimescaleDB version affected

2.6.0-dev

### PostgreSQL version used

15devel

### What operating system did you use?

Ubuntu 20.04.3 LTS

### What installation method did you use?

Source

### What platform did you run on?

Not applicable

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

_No response_

### PR DESCRIPTION
**Pull request by mkindahl**
Starting with PG15, default permissions on the public schema is
restricted for any non-superuser non-owner. This causes test failures
since tables can no longer be created without explicitly adding
permissions, so we remove grant when bootstrapping the data nodes and
instead grant permissions to the users in the regression tests. This
keeps the default permissions on data nodes, but allow regression tests
to run.

Fixes #3957

Reference: https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=b073c3cc

### CODE REVIEW

**Review by mfundul — APPROVED** [review]

File: test/runner.sh [hunk 1]
```diff
@@ -59,9 +59,9 @@ trap cleanup EXIT
 # we use mkdir here because it is an atomic operation unlike existance of a lockfile
 # where creating and checking are 2 separate operations
 if mkdir ${TEST_OUTPUT_DIR}/.pg_init 2>/dev/null; then
-  cat <<EOF | ${PSQL} "$@" -U ${USER} -d template1 -v ECHO=none >/dev/null 2>&1
+  ${PSQL} "$@" -U ${USER} -d template1 -v ECHO=none >/dev/null 2>&1 <<EOF
     SET client_min_messages=ERROR;
-    GRANT CREATE ON SCHEMA public TO PUBLIC;
+    GRANT ALL ON SCHEMA public TO PUBLIC;
```
Comment (svenklemm) [inline test/runner.sh:64]:
```suggestion
    GRANT CREATE ON SCHEMA public TO ${TEST_PGUSER};
    GRANT CREATE ON SCHEMA public TO ${TEST_ROLE_DEFAULT_PERM_USER};
    GRANT CREATE ON SCHEMA public TO ${TEST_ROLE_DEFAULT_PERM_USER_2};
    GRANT CREATE ON SCHEMA public TO ${TEST_ROLE_1};
    GRANT CREATE ON SCHEMA public TO ${TEST_ROLE_2};
    GRANT CREATE ON SCHEMA public TO ${TEST_ROLE_3};
```
You are still modifying the default ACL here which I think should be avoided and instead we should adjust the permissions of the actual users we use here.

**Review by svenklemm — CHANGES_REQUESTED** [review]
You are still modifying the default ACL here which I think should be avoided and instead we should adjust the permissions of the actual users we use here.

**Comment by afiskon** [comment]
> Commit #3941 fixed this by granting full permissions to the public
schema to all users when bootstrapping a data node, but this means that
non-superusers have full access to public schema on the data node,
which is opposite to the intention of the fix in PG15.

I would like to point out that this part of the commit message is inaccurate. What the commit does is grant the permissions to the users (!) who have access to the given database, not to all users. These are the default permissions in PG <= 15.

File: test/runner_shared.sh [hunk 2]
```diff
@@ -38,7 +38,10 @@ shift
 # we use mkdir here because it is an atomic operation unlike existance of a lockfile
 # where creating and checking are 2 separate operations
 if mkdir ${TEST_OUTPUT_DIR}/.pg_init 2>/dev/null; then
-  ${PSQL} "$@" -U ${USER} -d postgres -v ECHO=none -c "ALTER USER ${TEST_ROLE_SUPERUSER} WITH SUPERUSER;" >/dev/null
+  ${PSQL} "$@" -U ${USER} -d postgres -v ECHO=none >/dev/null <<EOF
+GRANT ALL ON SCHEMA public TO PUBLIC;
```
Comment (fabriziomello) [inline test/runner_shared.sh:42]: Shouldn't we do here the same you did in the `runner.sh`?

**Comment by mkindahl** [comment]
> > Commit #3941 fixed this by granting full permissions to the public
> > schema to all users when bootstrapping a data node, but this means that
> > non-superusers have full access to public schema on the data node,
> > which is opposite to the intention of the fix in PG15.
> 
> I would like to point out that this part of the commit message is inaccurate. What the commit does is grant the permissions to the users (!) who have access to the given database, not to all users. These are the default permissions in PG < 15.

Updated comment to remove that part of the message. Just focusing on the changes of the commit.

File: test/runner_shared.sh (hunk 2 above)
Comment (mkindahl) [reply test/runner_shared.sh]: It is not necessary here, so would like to re-visit this separately since it is not critical for this fix.

**Comment by mkindahl** [comment]
> You are still modifying the default ACL here which I think should be avoided and instead we should adjust the permissions of the actual users we use here.

Well... the default ACL setting is moved from the code to the tests, so strictly speaking I am not "still" modifying the default ACL.

Given our discussion before, `trusted_extension` does have a problem so switched this to grant individual permissions to the users we have and also extended that test with a negative test, which was missing.

**Review by svenklemm — APPROVED** [review]

**Review by fabriziomello — APPROVED** [review]

**Pull request merged**